custom_decorator/activity_utils.py

```diff
@@ -0,0 +1,32 @@
+import asyncio
+from datetime import datetime
+from functools import wraps
+from typing import Any, Awaitable, Callable, TypeVar, cast
+
+from temporalio import activity
+
+F = TypeVar("F", bound=Callable[..., Awaitable[Any]])
+
+
+def auto_heartbeater(fn: F) -> F:
+    # We want to ensure that the type hints from the original callable are
+    # available via our wrapper, so we use the functools wraps decorator
+    @wraps(fn)
+    async def wrapper(*args, **kwargs):
+        # Heartbeat twice as often as the timeout
+        heartbeat_timeout = activity.info().heartbeat_timeout
+        if heartbeat_timeout:
+            asyncio.create_task(heartbeat_every(heartbeat_timeout.total_seconds() / 2))
```

Comment:
When would this stop heartbeating?

Reply:
Only on cancel which is fine for this demonstration but not for general purpose use. Per comment below, I will fix.

Reply:
Fixed